### PR TITLE
Basic parquet implementation

### PR DESCRIPTION
**Pull request by jakobbraun**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.stream.Collectors;
+
+import com.exasol.adapter.document.documentnode.DocumentNode;
+import com.exasol.adapter.document.documentnode.DocumentObject;
+
+/**
+ * {@link DocumentNode} wrapping a java map.
+ */
+public class JavaObjectNode implements DocumentObject {
+    /** @serial */
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java:13]: Same here

File: src/test/java/com/exasol/adapter/document/documentfetcher/files/AbstractFilesDocumentFetcherTest.java [hunk 2]
```diff
@@ -30,10 +30,10 @@ void testSourceReferences() {
         when(fileLoader.loadFiles()).thenReturn(Stream.of(mockLoadedFile("file-1"), mockLoadedFile("file-2")));
         final FileLoaderFactory loaderFactory = mock(FileLoaderFactory.class);
         when(loaderFactory.getLoader(Mockito.any(), Mockito.any(), Mockito.any())).thenReturn(fileLoader);
-        final AbstractFilesDocumentFetcher<Object> documentFetcher = mock(AbstractFilesDocumentFetcher.class,
+        final AbstractFilesDocumentFetcher documentFetcher = mock(AbstractFilesDocumentFetcher.class,
```
Comment (AnastasiiaSergienko) [inline src/test/java/com/exasol/adapter/document/documentfetcher/files/AbstractFilesDocumentFetcherTest.java:33]: Maybe if you use class fields with Mock annotations these two methods would be easier to read

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]
It looks good, I have a few small suggestion

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.stream.Collectors;
+
+import com.exasol.adapter.document.documentnode.DocumentArray;
+import com.exasol.adapter.document.documentnode.DocumentNode;
+
+/**
+ * This class wraps list of java objects as {@link DocumentNode}.
+ */
+public class JavaListNode implements DocumentArray {
+    /** @serial */
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java:13]: :+1:

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private final List<Object> values;
+
+    public JavaListNode(final List<Object> values) {
+        this.values = values;
+    }
+
+    @Override
+    public List<? extends DocumentNode> getValuesList() {
+        return this.values.stream().map(JavaObjectDocumentNodeFactory::getNodeFor).collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java:22]: It should only be called once

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public DocumentNode getValue(final int index) {
+        return JavaObjectDocumentNodeFactory.getNodeFor(this.values.get(index));
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/documentnode/avro/JavaListNode.java:27]: same

File: src/test/java/com/exasol/adapter/document/documentfetcher/files/AbstractFilesDocumentFetcherTest.java (hunk 2 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/document/documentfetcher/files/AbstractFilesDocumentFetcherTest.java:33]: yes. I refactored it. What do you think?

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java [hunk 6]
```diff
@@ -10,6 +10,7 @@
  * {@link DocumentNode} wrapping a java map.
  */
 public class JavaObjectNode implements DocumentObject {
+    private static final long serialVersionUID = 1980967934702336379L;
     /** @serial */
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java:14]: Please check other classes with /** @ serial */
I think UID can be missing in some other places too

**Review by AnastasiiaSergienko — COMMENTED** [review]
The build is red, check it please

File: src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java (hunk 6 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/document/documentnode/avro/JavaObjectNode.java:14]: done

File: pom.xml [hunk 7]
```diff
@@ -4,9 +4,9 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.exasol</groupId>
     <artifactId>virtual-schema-common-document-files</artifactId>
-    <version>2.0.0</version>
+    <version>2.0.0-SNAPSHOT</version>
     <name>Virtual Schema for document data in files</name>
-    <description>Adapter for document data access from files.</description>
+    <description>Adapter for document data access from filParquetDocumentFetcherTestes.</description>
```
Comment (AnastasiiaSergienko) [inline pom.xml:9]: Is it some copy-paste error?

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with one editorial finding

**Pull request merged**